service/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -132,11 +135,14 @@ private List<DatasetResponse> collectDatasets(StorageSystem system) {
     return createDatasetResponses(roleMap, system);
   }
 
-  public DatasetsListResponse listDatasets() {
+  public DatasetsListResponse listDatasets() throws InterruptedException {
     List<DatasetResponse> datasets = new ArrayList<>();
+    ExecutorService es = Executors.newCachedThreadPool();
```

Comment:
Can you use a parallel stream instead of creating an executor service?
Something like
```
List<DatasetResponse> datasets = Arrays.stream(StorageSystem.values()).parallel().map(this::collectDatasets).toList();
```